spec/Converter/ValueConverterSpec.php

```diff
@@ -0,0 +1,225 @@
+
+    private const INTEGER_ATTRIBUTE_CODE = 'position';
+
+    private const SELECT_ATTRIBUTE_CODE = 'select';
+
+    private const DATETIME_ATTRIBUTE_CODE = 'created_at';
+
+    private const PRODUCT_OPTION_CODE = 'finitura';
+
+    const IT_LOCALE_CODE = 'it_IT';
+
+    public function let(
+        TranslatorInterface $translator,
+        AttributeInterface $textAttribute,
+        AttributeInterface $checkboxAttribute,
+        AttributeInterface $textareaAttribute,
+        AttributeInterface $integerAttribute,
+        AttributeInterface $selectAttribute,
+        AttributeInterface $datetimeAttribute
+    ) {
+        $translator->trans('webgriffe_sylius_akeneo.ui.value_converter', ['unit' => 'INCH', 'amount' => '23'], null, 'it')->willReturn('23"');
+
+        $textAttribute->getType()->willReturn(TextAttributeType::TYPE);
+        $integerAttribute->getType()->willReturn(IntegerAttributeType::TYPE);
+        $textareaAttribute->getType()->willReturn(TextareaAttributeType::TYPE);
+        $checkboxAttribute->getType()->willReturn(CheckboxAttributeType::TYPE);
+        $selectAttribute->getType()->willReturn(SelectAttributeType::TYPE);
+        $datetimeAttribute->getType()->willReturn(DatetimeAttributeType::TYPE);
+
+        $selectAttribute->getConfiguration()->willReturn(
+            [
+                'choices' => [
+                    'brand_agape_IT' => ['it_IT' => 'Agape Italia', 'en_US' => 'Agape Italy'],
+                    'brand_agape_US' => ['it_IT' => 'Agape USA', 'en_US' => 'Agape US'],
+                    'brand_agape' => ['it_IT' => 'Agape', 'en_US' => 'Agape'],
+                    'brand_agape_plus' => ['it_IT' => 'Agape Pus', 'en_US' => 'Agape Plus'],
+                ],
+            ]
+        );
+
+        $this->beConstructedWith();
+    }
+
+    function it_is_initializable()
+    {
+        $this->shouldHaveType(ValueConverter::class);
+    }
+
+    function it_implements_value_handler_interface()
+    {
+        $this->shouldHaveType(ValueConverterInterface::class);
+    }
+
+    function it_throws_exception_during_convert_when_value_not_contains_amount_key(
+        AttributeInterface $textAttribute
+    ) {
+        $this->shouldThrow(
+            new \LogicException('Amount key not found')
+        )->during('convert', [
+            $textAttribute,
+            [
+                'key' => 'value',
+            ],
+            'it',
+        ]);
+    }
+
+    function it_throws_exception_during_convert_when_value_contains_amount_key_and_not_contains_unit_key(
+        AttributeInterface $textAttribute
+    ) {
+        $this->shouldThrow(
+            new \LogicException('Unit key not found')
+        )->during('convert', [
+            $textAttribute,
+            [
+                'amount' => 23,
+                'key' => 'value',
+            ],
+            'it',
+        ]);
+    }
+
+    function it_converts_metric_value_from_akeneo_to_text_value_not_translated_when_translator_is_not_injected(
```

Comment:
Why not always requiring the translator?

Reply:
To avoid generating a BC break in the AttributeValueHandler definition, we must allow the static implementation of the ValueConverter. To avoid injecting a secondary temporary service (the translator) we must allow the construction without this service. An alternative could be instantiate the service statically like this:
`$valueConverter = new ValueConverter(new IdentityTranslator());`

Reply:
You're right.